angel_system/interfaces/mm_detect_activities.py

```diff
@@ -0,0 +1,37 @@
+import abc
+from typing import Iterable, Hashable, Dict, Tuple, Any
+import numpy as np
+
+from smqtk_core import Configurable, Pluggable
+
+
+class MMDetectActivities (Configurable, Pluggable):
+    """
+    Algorithm that detects activities from a set of input
+    frames and multi-modal auxiliary data as iterables.
+    """
+
+    @abc.abstractmethod
+    def detect_activities(
+      self,
+      frame_iter: Iterable[np.ndarray],
+      aux_data_iter: Iterable[Dict]
```

Comment:
The `aux_data_iter` parameter here is pretty undefined and could be just about anything. In general, interfaces should have well-defined function signatures so that users know specifically what to provide to them, and also so that implementations know specifically what they are being provided. When things are too "loose" like this, users can easily provide incorrectly formatted inputs, and implementations can easily impose implementation-specific requirements that cause implicit coupling with using applications (e.g. no better than integrating/using the one specific implementation directly).

Due to reasons like the above, I would suggest not defining an interface specifically for your new approach and focusing on a stand-alone "implementation" to perform inference over a trained model. That implementation should be documented with specifically what it needs to perform its work. As we add more implementations/models that perform a similar input/output behavior, *then* would be a more appropriate time to create an interface as we have a better understanding of what is actually common among the implementations that we want to be hot-swappable.

Reply:
I have added some info to [this](https://github.com/PTG-Kitware/angel_system/pull/93#discussion_r926730668) comment. Will look into this comment once the linked comment is clarified.

Reply:
My comment here is that we should not have an interface at this time. We don't have multiple implementations to switch between, and we don't know the practical generalization of this based on practical application. We just, at a minimum, want to be able to simply perform forward inference passes on the model that is being trained.

Reply:
Cool. I have removed the `MMDetectActivities` interface for now.

Reply:
Is this overridden by the previous comment that states you kept the interface around? I assume yes since I still see the interface file in the diff.

Reply:
Oops. Yes I have kept the interface (anticipating future use) but am not using it currently.